pandasai/connectors/airtable.py

```diff
@@ -0,0 +1,243 @@
+        """
+        filename = self._get_cache_path(
+            include_additional_filters=self._additional_filters is not None
+            and len(self._additional_filters) > 0
+        )
+        df.to_parquet(filename)
+
+    @property
+    def fallback_name(self):
+        """
+        Returns the fallback table name of the connector.
+
+        Returns :
+            str : The fallback table name of the connector.
+        """
+        return self._config.table
+
+    def execute(self):
+        """
+        Execute the connector and return the result.
+
+        Returns:
+            DataFrameType: The result of the connector.
+        """
+        return self.fetch_data()
+
+    def build_formula(self):
+        """
+        Build Airtable query formula for filtering.
+        """
+
+        condition_strings = []
+        for i in self._config.where:
+            filter_query = f"{i[0]}{i[1]}'{i[2]}'"
+            condition_strings.append(filter_query)
+        filter_formula = f'AND({",".join(condition_strings)})'
+        return filter_formula
+
+    def fetch_data(self):
+        """
+        Feteches data from airtable server through
+            API and converts it to DataFrame.
+        """
+        url = f"{self._root_url}{self._config.base_id}/{self._config.table}"
+        params = {}
+
+        if self._config.where:
+            params["filterByFormula"] = self.build_formula()
+        response = requests.get(
+            url=url,
+            headers={"Authorization": f"Bearer {self._config.api_key}"},
+            params=params,
+        )
+        if response.status_code == 200:
+            data = response.json()
+            data = self.preprocess(data=data)
+            self._save_cache(data)
+        else:
+            raise InvalidRequestError(
+                f"""Failed to connect to Airtable. 
+                    Status code: {response.status_code}, 
+                    message: {response.text}"""
+            )
+        return data
+
+    def preprocess(self, data):
+        """
+        Preprocesses Json response data
+        To prepare dataframe correctly.
+        """
+        records = [
+            {"id": record["id"], **record["fields"]} for record in data["records"]
+        ]
+
+        df = pd.DataFrame(records)
+        return df
+
+    def head(self):
+        """
+        Return the head of the table that
+          the connector is connected to.
+
+        Returns :
+            DatFrameType: The head of the data source
+                 that the conector is connected to .
+        """
+        return self.fetch_data().head()
+
+    @property
+    def rows_count(self):
+        """
+        Return the number of rows in the data source that the connector is
+        connected to.
+
+        Returns:
+            int: The number of rows in the data source that the connector is
+            connected to.
+        """
+        data = self.execute()
```

Comment:
Keep properties of rows_count and store it in instance so we don't have to call again and again